### PR TITLE
doc(device-sample): Added readme and enhanced setup script for x509 cert chain sample

### PR DESCRIPTION
**Pull request by vinagesh**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# x509 device certificate with chain sample
+
+## How to run the sample
+
+1) Create IotHub and device to use in sample.
```
Comment (barustum) [inline iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md:5]: shouldn't creating a device be after you run the AuthSetup in order to grab the thumbprint that is generated by the script and use it when creating a device?

File: iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+-iotHubResourceGroup <IoT hub resource group> `
+-iotHubName <IoT hub name> `
+-deviceId <device name> 
+```
+
+## Important things to note
+
+1) Use the same IoT hub created in step 1 as input to step 2 and step 3.
+2) Use the same device created in step 1 as input to step 2 and step 3.
+3) Use the same password created in in step 2 as input to step 3.
```
Comment (barustum) [inline iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md:21]: When I ran the script and provided my own password, it complained that:
AuthSetup.ps1: Cannot process argument transformation on parameter 'rootCertPassword'. Cannot convert the "password" value of type "System.String" to type "System.Security.SecureString".

I had to create a plain password then call Convert-ToSecureString then pass that to the script.

Comment (vinagesh) [reply iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md:21]: Yes that is what is expected as it is a secure string input. But it would be good to call it out with an example in the readme.

File: iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+-certFolderPath <Path to root cert> `
+-rootCertPassword <root certificate password> `
+-iotHubResourceGroup <IoT hub resource group> `
+-iotHubName <IoT hub name> `
+-deviceId <device name> 
+```
+
+## Important things to note
+
+1) Use the same IoT hub created in step 1 as input to step 2 and step 3.
```
[the remark this answers is not in the document]
Comment (vinagesh) [reply iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md:19]: It's there below the powershell snippet. But let me add some details on inputs to step 3

File: iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md (hunk 1 above)
Comment (vinagesh) [reply iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md:5]: Not needed if you use the x.509 CA signed option. It is simpler to use that option. But both options will work fine. Though you are correct of the order of things if you use the self signed option.

File: iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (vinagesh) [reply iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md]: I updated the instructions some more to add clarity.

File: iot-hub/Samples/device/X509DeviceCertWithChainSample/Program.cs [hunk 4]
```diff
@@ -47,7 +47,6 @@ public static async Task<int> Main(string[] args)
 
             var sample = new X509DeviceCertWithChainSample(deviceClient);
             await sample.RunSampleAsync();
-            await deviceClient.CloseAsync();
```
Comment (bikamani) [inline iot-hub/Samples/device/X509DeviceCertWithChainSample/Program.cs:50]: why was this removed?

Comment (vinagesh) [reply iot-hub/Samples/device/X509DeviceCertWithChainSample/Program.cs:50]: It is already there in the sample run, not program.cs. I think someone added it here again and the sample was failing as we were calling Close twice.

**Review by barustum — APPROVED** [review]
